include/oneapi/dpl/pstl/hetero/dpcpp/unseq_backend_sycl.h

```diff
@@ -514,15 +514,24 @@ struct __scan
 
             // 1. reduce
             auto __k = 1;
+            auto __next_k = 2;
+            // TODO: use adjacent work items for better SIMD utilization
+            // Consider the example with the mask of work items performing reduction:
+            // iter    now         proposed
+            // 1:      10101010    11110000
```

Comment:
Won't we break associativity here?

Reply:
It is proposed to change alignment of work items. Positions of elements to handle remain the same as well as reduction order, so this does not affect associativity.